service/matching/matchingEngine.go

```diff
@@ -152,20 +152,23 @@ func (e *matchingEngineImpl) String() string {
 	return r
 }
 
-// Returns taskListManager for a task list. If not already cached gets new range from DB and if successful creates one.
 func (e *matchingEngineImpl) getTaskListManager(taskList *taskListID) (taskListManager, error) {
-	e.taskListsLock.RLock()
-	if result, ok := e.taskLists[*taskList]; ok {
-		e.taskListsLock.RUnlock()
-		return result, nil
-	}
-	e.taskListsLock.RUnlock()
-	mgr := newTaskListManager(e, taskList, e.config)
+	return e.getTaskListManagerWithRPS(taskList, nil)
+}
+
+// Returns taskListManager for a task list. If not already cached gets new range from DB and
+// if successful creates one. The passed in throttling limit determines how many tasks are
+// dispatcher per second.
+func (e *matchingEngineImpl) getTaskListManagerWithRPS(
+	taskList *taskListID, maxDispatchPerSecond *float64,
+) (taskListManager, error) {
 	e.taskListsLock.Lock()
 	if result, ok := e.taskLists[*taskList]; ok {
+		result.UpdateMaxDispatch(maxDispatchPerSecond)
```

Comment:
if aren't gonna switch to atomics, consider moving this outside the lock so that, we don't block addition of new task lists, when another tasklist is blocked on UpdateMaxDispatch for multiple seconds if/when they roll out a change in poll limit